packages/backend-deployer/src/cdk_deployer.ts

```diff
@@ -39,6 +39,8 @@ export class CDKDeployer implements BackendDeployer {
     if (deployProps?.deploymentType === 'sandbox') {
       cdkCommandArgs.push('--hotswap-fallback');
       cdkCommandArgs.push('--method=direct');
+      // The context will be consumed by data construct to allow destructive updates and replacing table upon GSI change
+      cdkCommandArgs.push('--context', 'amplifySandboxDeploy=true');
```

Comment:
There's already information about this, see https://github.com/aws-amplify/amplify-backend/blob/70685f36bdd089711fab3ba6eab0c375dad35827/packages/platform-core/src/cdk_context_key.ts#L10 .

But, before you encode this modality in grapqhql construct we should sync up with folks who work on similar thing for auth deletion protection.

@srquinn21 @edwardfoyle @Amplifiyer thoughts ?

Reply:
The concept of the sandbox shouldn't leak to the L3 constructs. We should handle this in a layer up (possibly like `backend-data` package as @sobolk suggested) and we should do it with a CDK `Aspect` which is what we plan to do with other categories like auth. This allows the constructs to remain agnostic to Sandbox's existence and constrain the modality to one place. I do not think we should add this context flag as its a 1 way door if we do the above. Seems like we need to hash out a design here.

Reply:
Even more than that, perhaps we should take the approach that we are taking with auth: The construct itself should not be responsible for trying to protect the customer from accidental deletion -- that should be the job of the CI/CD pipeline to make sure you aren't making a mistake before deploying to a real environment. This would mean that the construct will _always_ delete/recreate/whatever is necessary to get the cloud in sync with the code definition.

I am willing to concede that the data construct and the rest of samsara may have slightly different design goals here though. If that's the case, then we should do what @sobolk  and @srquinn21 are suggesting and have the `backend-data` layer set these values on the construct

Reply:
Hey folks, sorry I didn't see this thread until just now. Yeah factory should pull whatever context flags are available, and can pass those into CDK construct via predefined `allowDestructiveGraphqlupdates` and `replaceOnGSIChange` feature flags.